tests/trainer/test_trainer.py

```diff
@@ -638,23 +638,31 @@ def test_trainer_max_steps_accumulate_batches(tmpdir):
     assert trainer.global_step == trainer.max_steps, "Model did not stop at max_steps"
 
 
-def test_benchmark_option(tmpdir):
+@pytest.mark.parametrize(
+    ["benchmark", "deterministic", "expected"],
+    [
+        (None, False, True),
+        (None, True, False),
+        (True, False, True),
+        (True, True, True),
+        (False, True, False),
+        (False, False, False),
+    ],
+)
+def test_benchmark_option(benchmark, deterministic, expected):
     """Verify benchmark option."""
 
-    model = BoringModel()
-
-    # verify torch.backends.cudnn.benchmark is not turned on
-    assert not torch.backends.cudnn.benchmark
+    original_val = torch.backends.cudnn.benchmark
 
-    # fit model
-    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, benchmark=True)
-    trainer.fit(model)
-
-    # verify training completed
-    assert trainer.state.finished, f"Training failed with {trainer.state}"
+    if benchmark and deterministic:
+        with pytest.warns(RuntimeWarning, match="You passed `deterministic=True` and `benchmark=True`"):
+            trainer = Trainer(benchmark=benchmark, deterministic=deterministic)
+    else:
+        trainer = Trainer(benchmark=benchmark, deterministic=deterministic)
+    assert torch.backends.cudnn.benchmark == expected
+    assert trainer.benchmark == expected
```

Comment:
```suggestion
    assert trainer._accelerator_connector.benchmark == expected
```
oh! it's accelerator connector.